mosaic/read_mosaic.c

```diff
@@ -259,13 +259,13 @@ void get_var_data(const char *file, const char *name, void *data)
   switch (vartype) {
     case NC_DOUBLE:case NC_FLOAT:
 #ifdef OVERLOAD_R4
-      status = nc_get_var_float(ncid, varid, data);
+      status = nc_get_var_float(ncid, varid, (float *)data);
```

Comment:
Is this OVERLOAD potentially affecting how things work and should the NC_DOUBLE and NC_FLOAT cases be broken into individual cases?

Reply:
These changes were put in by seth in #654 actually to fix some compile issues on darwin, but they look good to me. With the previous logic it would read in data as float or double based on whichever is the fortran default kind so this just adds the cast to data since its a void pointer.